tiledb/appl/blob_array/test/unit_blob_array_main.cc

```diff
@@ -0,0 +1,607 @@
+
+  basic_to_array(false);
+
+  // array open in READ mode, but EMPTY
+  // And, array is empty since any array writes above (should have) failed.
+  basic_export(false);
+
+  REQUIRE(blob_array->close().ok() == true);
+
+  open_for_write();
+
+  // since empty, nothing to export
+  basic_export(false);
+
+  // array open WRITE but empty (unless #ifdef code above active)
+  basic_to_array(true);
+
+  REQUIRE(blob_array->close().ok() == true);
+
+  open_for_read();
+
+  // array open for read, has something in it
+  basic_to_array(false);  // open read, unable to add
+
+  {
+    std::string encryption_type_string =
+        encryption_type_str((tiledb::sm::EncryptionType)encryption_type);
+    std::cout << "b4 export, encryption_type " << encryption_type_string
+              << std::endl;
+  }
+  basic_export(true);  // open for read and non-empty, should succeed
+
+  REQUIRE(blob_array->close().ok() == true);
+
+  tiledb::appl::WhiteboxBlobArray wb_ba(*blob_array);
+  auto open_wbba_for_write = [&]() -> void {
+    if (wb_ba.is_open())
+      REQUIRE(wb_ba.close().ok() == true);
+    auto open_res = wb_ba.open(
+        tiledb::sm::QueryType::WRITE,
+        encryption_type,
+        encryption_key,
+        static_cast<uint32_t>(
+            strlen(encryption_key)));  // inherited from parent
+    REQUIRE(wb_ba.is_open() == true);
+  };
+  auto open_wbba_for_read = [&]() -> void {
+    if (wb_ba.is_open())
+      REQUIRE(wb_ba.close().ok() == true);
+    auto open_res = wb_ba.open(
+        tiledb::sm::QueryType::READ,
+        encryption_type,
+        encryption_key,
+        static_cast<uint32_t>(strlen(encryption_key)));
+    REQUIRE(wb_ba.is_open() == true);
+  };
+
+  tiledb::sm::Datatype value_type;
+  uint32_t value_size;
+  const void* ptr_value;
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      false);
+  open_wbba_for_write();
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == false);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      false);
+  open_wbba_for_read();
+  CHECK(wb_ba.get_file_ext(&value_type, &value_size, &ptr_value).ok() == true);
+  CHECK(wb_ba.get_mime_type(&value_type, &value_size, &ptr_value).ok() == true);
+  CHECK(
+      wb_ba.get_mime_encoding(&value_type, &value_size, &ptr_value).ok() ==
+      true);
+
+  REQUIRE(blob_array->close().ok() == true);
+
+  auto show_dir = [&](std::string path) {
+    std::vector<tiledb::sm::URI> filelist;
+    tiledb::sm::URI uri_path(path);
+    vfs_->vfs_->ls(uri_path, &filelist);
+    std::cout << "path " << path << ", nitems " << filelist.size() << std::endl;
```

Comment:
remove

Reply:
removed (tho useful header summary of info following)